themes/default/views/editor/objects/screen_html.php

```diff
@@ -82,4 +82,58 @@
 
 	<div class="editorBottomPadding"><!-- empty --></div>
 	
-	<?php print caSetupEditorScreenOverlays($this->request, $t_object, $va_bundle_list); ?>
\ No newline at end of file
+	<?php print caSetupEditorScreenOverlays($this->request, $t_object, $va_bundle_list); ?>
+	<script type="text/javascript">
+		jQuery(function () {
+			var parse = function (val) {
+				if (typeof val === 'string') {
+					val = val.trim();
+					if (val) {
+						var parts = val.match(/^\.?\s?(\d*)\s*(.*)$/);
+						parts.shift();
+						return parts;
+					}
+				}
+				return undefined;
+			};
+			jQuery('#idno_accession_series_and_parts').each(function () {
+				var $dot = jQuery('<span class="idno_dot">.</span>').hide(),
+					$series = jQuery('<input type="number" class="idno_accession_series" title="Enter series number" step="1" min="1"/>'),
+					$suffix = jQuery('<input type="text" class="idno_accession_suffix" size="10"/>'),
+					$this = jQuery(this).hide().before($dot).before($series).before($suffix),
+					seriesAndParts = parse($this.val()),
+					events = 'keyup change',
+					serialize = function () {
+						var ret = '',
+							series = $series.val().trim();
+						if (series) {
+							$dot.show();
+							ret += '.' + series;
+						} else {
+							$dot.hide();
+						}
+						ret += $suffix.val().trim();
+						$this.val(ret);
+					};
+				if (seriesAndParts) {
+					if (seriesAndParts[0] >= 1) {
+						$dot.show();
+						$series.val(seriesAndParts[0]);
+					}
+					$suffix.val(seriesAndParts[1]);
+				}
+				var $numeric = jQuery('#idno_accession_year,#idno_accession_number,.idno_accession_series');
```

Comment:
I decided to swallow all nonnumeric character input in all numeric fields. The link on SO was a bit too greedy - I couldn't tab out of the field once I'd tabbed into it, and the only character below 48 that had any adverse effect on the field is space (up and down to change values, enter to submit form, backspace / delete to fix errors, and the rest are pretty much innocuous).